qiskit/circuit/library/standard_gates/x.py

```diff
@@ -181,6 +181,7 @@ def __init__(self, label=None, ctrl_state=None):
         super().__init__(
             "cx", 2, [], num_ctrl_qubits=1, label=label, ctrl_state=ctrl_state, base_gate=XGate()
         )
+        self._qasm3_definition = "gate cx c, t {CX c, t;}\n"
```

Comment:
My understanding is this is temporary for CX, but do you know how we're planning to handle this generically for other gates?

Reply:
The specific problem with `cx` is that now, `cx` is not a "primitive" gate (like `U`). And, in qiskit, there is no `_define` for `cx`. So, if the dump cannot use the standard include, `cx` is opaque. `_qasm3_definition` is  there to handle that.

@jakelishman has a longer explanation here https://github.com/Qiskit/qiskit-terra/pull/6565/files#r722398716